src/components/Modal_Components/demomanualBlocks.js

```diff
@@ -1,383 +1,156 @@
-import { Modal_Title } from "./Modal_Title";
-import MuiDialogActions from "@material-ui/core/DialogActions";
-import { addManualBlocks,editManualBlocks } from '../../redux/actions/ManualActions';
+import React, { useState } from "react";
+import { useStyles } from "./dialogViews/dialogStyles";
+import {
+  addManualBlocks,
+  editManualBlocks
+} from "../../redux/actions/ManualActions";
+import {
+  dialogView1,
+  dialogView2,
+  dialogView3,
+  dialogView4
+} from "./dialogViews";
 
-import { withStyles, makeStyles } from "@material-ui/core/styles";
-import Button from "@material-ui/core/Button";
-import Dialog from "@material-ui/core/Dialog";
-import MuiDialogContent from "@material-ui/core/DialogContent";
 import { connect } from "react-redux";
-
-import FormControl from "@material-ui/core/FormControl";
-import TextField from "@material-ui/core/TextField";
-import Select from "@material-ui/core/Select";
+import "./dialogViews/dialogComponents";
 import "./modalStyle.css";
-const useStyles = makeStyles(theme => ({
-  root: {
-    width: "100%",
-    display: "flex",
-    justifyContent: "space-between"
-  },
-  formControl: {
-    width: "45%"
-  }
-}));
 
-const DialogContent = withStyles(theme => ({
-  root: {
-    padding: theme.spacing(2)
-  }
-}))(MuiDialogContent);
-const DialogActions = withStyles(theme => ({
-  root: {
-    margin: 0,
-    padding: theme.spacing(1)
-  }
-}))(MuiDialogActions);
 export const AddTransaction = props => {
-  const [customBlock,setCustomBlock] = useState({name:"",budget:null,total:0})
-  const [editBlock,setEditBlock] = useState({name:"",budget:null,categoryid:null,total:0})
-  const [failedEdit,setFailedEdit] = useState(false)
-  const [failedCustom,setFailedCustom] = useState(false)
-  const [edit,setEdit] = useState(true)
+  const [state, setState] = useState({
+    customBlock: {
+      name: "",
+      budget: null,
+      total: 0
+    },
+    editBlock: {
+      name: "",
+      budget: null,
+      categoryid: null,
+      total: 0
+    },
+    edit: true,
+    failedEdit: false,
+    failedCustom: false
+  });
   const classes = useStyles();
-
+  const options = [{ label: "Create Block" }];
 
   const handleChange = e => {
-    
-    setCustomBlock({...customBlock, [e.target.name]: e.target.value});
-    setEditBlock({...editBlock,name:""})
-}
-const handleEdit = e => {
-    if (cat[e.target.value][1].name === "Create New Category")
-    {setEdit(false)}
-    else{setEdit(true)}
-    setEditBlock({...editBlock,name:cat[e.target.value][1].name,categoryid:e.target.value,budget:customBlock.budget})
-    setCustomBlock({...customBlock,name:""})
-}
-const handleBudget = e => {
-    setEditBlock({...editBlock,budget:e.target.value})
-    setCustomBlock({...customBlock,budget:e.target.value})
-    console.log(editBlock)
-    console.log(customBlock)
-}
-
-const handleSubmit = e => {
-    e.preventDefault()
-    if (customBlock.name === null || isNaN(customBlock.budget) || !isNaN(customBlock.name)){
```

Comment:
Good job at pulling this out into a variable on 67-70. Made it much more readable than this giant conditional line.